api/openapi.yaml

```diff
@@ -0,0 +1,327 @@
+        - request_id
+
+    Task:
+      type: object
+      properties:
+        description:
+          type: string
+          example: "an example task"
+        enabled:
+          type: boolean
+          example: true
+          default: true
+        name:
+          type: string
+          example: "taskA"
+        providers:
+          type: array
+          items:
+            type: string
+          example: []
+        services:
+          type: array
+          items:
+            type: string
+            example: ["web", "api"]
+        source:
+          type: string
+          example: "org/example/module"
+        variables:
+          type: object
+          additionalProperties:
+            type: string
+        version:
+          type: string
+          example: "1.0.0"
+        working_dir:
+          type: string
+          example: "./sync-tasks/task-name"
+          default: "sync-tasks"
+        buffer_period:
+          type: object
+          properties:
+            enabled:
+              type: boolean
+              example: true
+            max:
+              type: string
+              example: "5s"
+            min:
+              type: string
+              example: "20s"
+        condition:
+          type: object
+          properties:
+            catalog-services:
+              $ref: '#/components/schemas/CatalogServicesCondition'
+            services:
+              $ref: '#/components/schemas/ServicesSourceInput'
+            consul-kv:
+              $ref: '#/components/schemas/ConsulKVCondition'
+            schedule:
+              $ref: '#/components/schemas/ScheduleCondition'
+        source_input:
+          type: object
+          properties:
+            services:
+              $ref: '#/components/schemas/ServicesSourceInput'
+            consul-kv:
+              $ref: '#/components/schemas/ConsulKVSourceInput'
+      required:
+        - name
+        - source
+      example:
+        condition:
+          catalog-services:
+            regexp: ".*"
+        description: ''
+        enabled: true
+        name: taskC
+        providers: [ ]
+        services:
+          - web
+          - api
+        source: org/example/module
+        variable_files: [ ]
+        version: 1.0.0
+
+    CatalogServicesCondition:
+      type: object
+      properties:
+        datacenter:
+          type: string
+          example: "dc1"
+        regexp:
+          type: string
+          example: "web.*"
+        namespace:
+          type: string
+          example: "default"
+        node_meta:
+          type: object
+          additionalProperties:
+            type: string
+          example:
+            key: value
+        source_includes_var:
+          type: boolean
+          example: false
+    ConsulKVCondition:
+      type: object
+      properties:
+        path:
+          type: string
+          example: "my-key"
+        recurse:
+          type: boolean
+          example: false
```

Comment:
The default for `recurse` is false. Would we want to include the default value for this for consistency?